### ISSUE TITLE
Obsolete attribute on overridden virtual getter/setter gives no warning

### ISSUE THREAD

**Issue by SpaceOgre** [dotnet/roslyn#81764]
**Version Used**: 10.0.0

**Steps to Reproduce**:

1. Create a base class with a property
2. Inherit from the base class and mark the properties get or set as Obsolete
3. No warning is given

https://lab.razor.fyi/#lZDfSgJBFMaRAmuuoic47JXebKB3LV6oQRRZQkZQBJ3dPW2DuzM2Z1YT8Yl6A9-mm-57gzD_tGpazsXAnO_jN9_5xCgvRNPoyGDiBnz4lk9Zqgiu-2wp8UT25dZ1HFNgpVbsnpIiI4MNjrMkSS36MS15TiRGSrOVAa9X3LoOqaow7rNctl1I9bI0aj0bwlCqaN3cbSG32RMiiJEZphuLgQAAYItWBtDVMoQGSlUofo8n4vh00QBCBRT1oFooenMB3abRHTK2DxVwWsTW8eDo6FJDD42SKhILDH_KqGUZ_u-MBpo2hYAMIXUMBWgpBD-1oH7gq4jyAuM24xuKoRCd1I9lAJMSasg0aWA67kpjU4yBrRl3OI81gIisBzy-VjFVOM6gZizdJWNkSOtg91c-65gsFZwbprlcAqnYEoZO8WH2YYa6BCv9Ea22GO0ff26zwtbxNljLW_QyX_Z8x6Tqbn9vNPp8_3g62H3Mvea-AA

**Diagnostic Id**: CS0619

**Expected Behavior**:
I would expect to see a warning, like I do for none inherited properties.

**Actual Behavior**:
No warning recived

**Comment by jjonescz** [comment]
Pasting the snippet here for future reference:

```cs
using System;
using System.Collections.Generic;
using System.Collections.Immutable;
using System.Diagnostics;
using System.Diagnostics.CodeAnalysis;
using System.Linq;
using System.Threading;
using System.Threading.Tasks;

class Program
{
    static void Main()
    {
        var a = new A();
        a.Property = "Test"; //No warning

        var b = new B();
        b.Property = "Test"; //Marked as deprecated but no warning
        b.Property3 = "Test"; //Warning
    }
}

public class Base
{
   public virtual string Property { get; set; }
}

public class A : Base
{
    public override string Property { get; [Obsolete("Use Property2 instead")] set; }
    public string Property2 { get; set; }
}

public class B : Base
{
    [Obsolete("Use Property2 instead")]
    public override string Property { get; set; }
    public string Property2 { get; set; }

    public string Property3 { get; [Obsolete("Use Property2 instead")]set; }
}
```

This is by design. `b.Property` binds to `A.Property` per C# language rules. That's also why you are getting warning CS0809 on the `B.Property` declaration. See https://learn.microsoft.com/en-us/dotnet/csharp/misc/cs0809#methods-recognized-as-obsolete.
